checkmedia.c

```diff
@@ -1,527 +1,182 @@
-    if(*s != 0 && *s != ' ') break;
+  if(mediacheck_digest_valid(media->digest.part)) {
+    log_info(
+      "part %6s: %s\n",
+      mediacheck_digest_name(media->digest.part),
+      mediacheck_digest_hex(media->digest.part)
+    );
   }
 
-  if(!strncmp(iso.app_id, "MKISOFS", sizeof "MKISOFS" - 1)) *iso.app_id = 0;
-
-  if((s = strrchr(iso.app_id, '#'))) *s = 0;
-
-  if(strstr(iso.app_id, "-DVD-")) iso.media_type = "DVD";
-
-  for(s = iso.vol_id + sizeof iso.vol_id - 1; s >= iso.vol_id; *s-- = 0) {
-    if(*s != 0 && *s != ' ') break;
+  if(mediacheck_digest_valid(media->digest.full)) {
+    log_info(
+      "%s: %s\n",
+      mediacheck_digest_name(media->digest.full),
+      mediacheck_digest_hex(media->digest.full)
+    );
   }
 
-  if(*iso.vol_id == 'S') {
-    idx = iso.vol_id[strlen(iso.vol_id) - 1];
-    if(idx >= '1' && idx <= '9') iso.media_nr = idx - '0';
-  }
+  ok = mediacheck_digest_ok(media->digest.iso) || mediacheck_digest_ok(media->digest.part);
 
-  if((s = strstr(iso.app_data, "md5sum="))) {
-    s += sizeof "md5sum=" - 1;
-    iso.digest.type = digest_md5;
-    iso.digest.size = MD5_DIGEST_SIZE;
-    iso.digest.name = "md5";
-  }
-  else if((s = strstr(iso.app_data, "sha1sum="))) {
-    s += sizeof "sha1sum=" - 1;
-    iso.digest.type = digest_sha1;
-    iso.digest.size = SHA1_DIGEST_SIZE;
-    iso.digest.name = "sha1";
-  }
-  else if((s = strstr(iso.app_data, "sha224sum="))) {
-    s += sizeof "sha224sum=" - 1;
-    iso.digest.type = digest_sha224;
-    iso.digest.size = SHA224_DIGEST_SIZE;
-    iso.digest.name = "sha224";
-  }
-  else if((s = strstr(iso.app_data, "sha256sum="))) {
-    s += sizeof "sha256sum=" - 1;
-    iso.digest.type = digest_sha256;
-    iso.digest.size = SHA256_DIGEST_SIZE;
-    iso.digest.name = "sha256";
-  }
-  else if((s = strstr(iso.app_data, "sha384sum="))) {
-    s += sizeof "sha384sum=" - 1;
-    iso.digest.type = digest_sha384;
-    iso.digest.size = SHA384_DIGEST_SIZE;
-    iso.digest.name = "sha384";
+  if(ok) {
+    dia_message("No errors found.", MSGTYPE_INFO);
   }
-  else if((s = strstr(iso.app_data, "sha512sum="))) {
-    s += sizeof "sha512sum=" - 1;
-    iso.digest.type = digest_sha512;
-    iso.digest.size = SHA512_DIGEST_SIZE;
-    iso.digest.name = "sha512";
+  else if(media->abort) {
+    dia_message("Media check canceled.", MSGTYPE_INFO);
   }
-
-  if(iso.digest.type) {
-    if(strlen(s) >= iso.digest.size) {
-      for(u = 0 ; u < iso.digest.size; u++, s += 2) {
-         if(sscanf(s, "%2x", &u1) == 1) {
-           iso.digest.old[u] = u1;
-         }
-         else {
-           break;
-         }
-      }
-      if(u == iso.digest.size) iso.digest.got_old = 1;
+  else {
+    if(media->err_block) {
+      sprintf(buf, "Error reading block %u.", media->err_block);
     }
+    else {
+      sprintf(buf, "Checksum wrong.");
+    }
+    sprintf(buf + strlen(buf), "\nThis medium is broken.");
+    dia_message(buf, MSGTYPE_ERROR);
```

Comment:
buf is used only in this context. Is possible to move definition here? As it is fixed size buffer, I feel safer if it is closer to its usage so harder to overlook if more or longer messages added to it.